.github/workflows/cover.yaml

```diff
@@ -23,7 +23,10 @@ jobs:
       with:
         go-version: '1.20'
     - run: "PATH=/usr/local/go/bin:$PATH make test-cover"
-    - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d # v3.1.4
+    - uses: Wandalen/wretry.action@master # Retry codecov upload. It is flaky due to a known issue https://community.codecov.com/t/upload-issues-unable-to-locate-build-via-github-actions-api/3954
```

Comment:
we shouldn't be using a github at a mutable ref (`master`) for security reasons, make sure you pin it to a specific commit and then comment the tag number next to it if there is one